src/main.cpp

```diff
@@ -111,19 +112,34 @@ int main(int argc, char **argv)
 	QFile f(path+"Preferences.ini");
 	f.open(QFile::ReadOnly);
 	QTextStream in (&f);
-	const QStringList content = in.readAll().split("\n");
+    const QStringList content = in.readAll().split("\n");
 	QString language="default";
-	for(auto s : content)
-	{
-		if(s.startsWith("language"))
-		{
-			language=s.split("=")[1];
+	for(auto s : content) {
+		if(s.startsWith("language")) {
+			language=s.split("=")[1].trimmed();
 		}
 	}
 
-	qDebug()<<language;
+	QString languageFileName;
+	QString osLanguage = QLocale::system().name().split("_")[0];
+	QDir directory(QCoreApplication::applicationDirPath()+"/resources/languages");
+	QStringList languages = directory.entryList(QStringList() << "*.qm",QDir::Files);
+	if(language == "auto"){
+		if(languages.contains("scopy_"+osLanguage+".qm")) {
+			languageFileName = QDir(QCoreApplication::applicationDirPath()+"/resources/languages/scopy_"+osLanguage+".qm").path();
+		}
+		else{
+			languageFileName = QDir(QCoreApplication::applicationDirPath()+"/resources/languages/scopy_en.qm").path();
+		}
+	}
+	else if(languages.contains(language+".qm")) {
+		languageFileName=QDir(QCoreApplication::applicationDirPath()+"/resources/languages/"+language+".qm").path();
+	}
+	else { 
+		languageFileName = language;
+	}
 
-	QString languageFileName=QDir(QCoreApplication::applicationDirPath()+"/resources/languages/"+language+".qm").path();
+	f.close();
 	myappTranslator.load(languageFileName);
```

Comment:
Wouldn't it be easier to just
```cpp
        if (translator->load(locale, QCoreApplication::applicationName(), "_", dir)) {
            app.installTranslator(translator);
        }
```
where dir would be the directory of the languages. In this way the line above will try to load `scopy_en` or `scopy_de`, any scopy_<language> depending on the language given by the locale. If no language is found in this directory then no translator should be installed.
The languages (*.qm files) should be compiled in a resource and attached to the executable. The current implementation won't work for flatpak and the paths to the translation files would need to be given manually.